db/version_set.cc

```diff
@@ -3563,6 +3563,7 @@ struct VersionSet::ManifestWriter {
         cfd(_cfd),
         mutable_cf_options(cf_options),
         edit_list(e) {}
+  ~ManifestWriter() { status.PermitUncheckedError(); }
```

Comment:
These are hard to tell because you could create a ManifestWriter, never do anything to it, and then destroy it -- that status would not be checked.  One alternative is to not have Status as member variables.